lib/grape/middleware/logger.rb

```diff
@@ -80,6 +81,10 @@ def after_failure(error)
     after(error[:status])
   end
 
+  def headers
+    env['grape.request.headers']
```

Comment:
Yes, if the request contains a ton of headers then there will be a ton of data here. It also currently only supports Grape. 

Any preferences on the specific headers you'd like here? Perhaps a configuration option with sensible defaults.